packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -35,13 +35,16 @@ module.exports = function s3 (config) {
     // @ts-ignore The `companion` property is added by middleware before reaching here.
     const client = req.companion.s3Client
 
-    if (!client || typeof config.bucket !== 'string') {
+    if (!client || (typeof config.bucket !== 'string' && typeof config.bucket !== 'function')) {
```

Comment:
Can you make a util `isValidBucket`?
```suggestion
    if (!client || !isValidBucket(config.bucket)) {
```

Reply:
sure

Reply:
After some thinking, we might not need such function if we use `getKey` and then validate that it has returned a non empty string.